learn/tutorials/vaadin-designer-tutorial/content.adoc

```diff
@@ -379,11 +384,103 @@ public class MainView extends PolymerTemplate<MainView.MainViewModel> {
     }
 }
 ----
-<1> A field for storing the service for future access
-<2> Spring passes in the service when the view is created
-<3> The passed reference to the service is stored in the field
-<4> Columns to grid are added and configured
-<5> A custom column is created for fetching the name of the company the contact belongs to.
-<6> All columns are configured to be sized based on the content of the columns
-<7> Items are fetched from the service and passed to the Grid
+<1> Add bean type as type parameter to the grid
+<2> A field for storing the service for future access
+<3> Spring passes in the service when the view is created
+<4> The passed reference to the service is stored in the field
+<5> Columns to grid are added and configured
+<6> A custom column is created for fetching the name of the company the contact belongs to.
+<7> All columns are configured to be sized based on the content of the columns
+<8> Items are fetched from the service and passed to the Grid
+
+Run the application. Now the Grid shows the contacts from the service.
+
+image::grid-populated-run-the-project.png[Grid configured and populated with items from the service.]
+
+[#java-enable-filtering]
+=== Enable filtering
+
+We would like to filter the grid based on first name and last name when the user types in a value in the filter textfield. For that, we need a value change listener to the textfield, and we need to pass the value of the text field to the contact service. The contact service and repository don't support filtering right now, so we need to add that support as well.
+
+Let's start with changing MainView.
+
+.`*MainView.java*`
+[source,java]
+----
+
+public class MainView extends PolymerTemplate<MainView.MainViewModel> {
+
+    // Fields omitted
+    ...
+    
+
+    public MainView(ContactService contactService) {
+        this.contactService = contactService;
+        // You can initialise any data required for the connected UI components here.
+        grid.addColumn(Contact::getFirstName).setHeader("First name");
+        grid.addColumn(Contact::getLastName).setHeader("Last name");
+        grid.addColumn(Contact::getEmail).setHeader("Email");
+        grid.addColumn(Contact::getStatus).setHeader("Status");
+        grid.addColumn(contact -> {
+            Company company = contact.getCompany();
+            return company == null ? "-" : company.getName();
+        }).setHeader("Company");
+        grid.getColumns().forEach(col -> col.setAutoWidth(true));
+
+        updateList(); <1>
+
+        filterText.setValueChangeMode(ValueChangeMode.LAZY); <2>
+        filterText.addValueChangeListener(e -> updateList()); <3>
+    }
+
+    private void updateList() {
+        grid.setItems(contactService.findAll(filterText.getValue())); <4>
+    }
+
+    // Template model omitted
+    ...
+}
+----
+
+<1> Extract the call to grid.setItems into a new method
```

Comment:
This extraction step was moved to the previous code example so the highlight is not needed here

Reply:
Fixed.